### PR TITLE
Add support for GestureArenaTeam captains.

### PR DESCRIPTION
**Pull request by amirh**
*I apologize for adding more metaphors here, please do let me know if you have a better way to express this than as a captain.*

The team captain wins the arena on behalf of the team.
When any of the team members claims victory for the arena the captain
accepts the gesture.

This is used when embeddeding platform views - we allow configuring a
set of gestures that should be forwarded to the platform view.
We add the set of gesture recognizers to a GestureArenaTeam with a
captain, and if the captain accepts the gesture we forward it to the
platform view.

### CODE REVIEW

**Comment by Hixie** [comment]
How does this differ from just putting the captain first in the list?

**Comment by amirh** [comment]
There are 2 ways for the team to win:

  1. There are no standing gesture arena members that are not part of the team: in this case the first member in the list wins.
  2. One of the team members claims victory.

In the second case, the member the claims that victory accepts the gesture, but with a captain, the captain will accept the gesture.

File: packages/flutter/lib/src/gestures/team.dart [hunk 1]
```diff
@@ -105,11 +110,24 @@ class _CombiningGestureArenaMember extends GestureArenaMember {
 /// the horizontal nor vertical drag recognizers can claim victory) the tap
 /// recognizer still actually wins, despite being in the team.
 ///
+/// [AndroidView] is using a team with a captain to decide which gestures are
```
Comment (Hixie) [inline packages/flutter/lib/src/gestures/team.dart]: nit: "is using" -> "uses"

File: packages/flutter/lib/src/gestures/team.dart [hunk 2]
```diff
@@ -105,11 +110,24 @@ class _CombiningGestureArenaMember extends GestureArenaMember {
...
+/// forwarded to the native view. For example if we want to forward taps and
+/// vertical scrolls to a native Android view, [TapGestureRecognizers] and
+/// [VerticalDragGestureRecognizer] are added to a team with a captain, once the
+/// captain wins then gesture is forwarded to the Android view.
```
Comment (Hixie) [inline packages/flutter/lib/src/gestures/team.dart]: this is good as far as it goes but it doesn't answer the question of why it uses a captain.

**Comment by Hixie** [comment]
LGTM

fyi @abarth

File: packages/flutter/lib/src/gestures/team.dart (hunk 1 above)
Comment (amirh) [reply packages/flutter/lib/src/gestures/team.dart]: done

File: packages/flutter/lib/src/gestures/team.dart (hunk 2 above)
Comment (amirh) [reply packages/flutter/lib/src/gestures/team.dart]: Tried to thicken the explanation a bit.

**Comment by amirh** [comment]
Thanks for the quick review!
Will land on green.

File: packages/flutter/lib/src/gestures/team.dart [hunk 3]
```diff
@@ -105,11 +110,26 @@ class _CombiningGestureArenaMember extends GestureArenaMember {
 /// the horizontal nor vertical drag recognizers can claim victory) the tap
 /// recognizer still actually wins, despite being in the team.
 ///
+/// [AndroidView] uses a team with a captain to decide which gestures are
+/// forwarded to the native view. For example if we want to forward taps and
+/// vertical scrolls to a native Android view, [TapGestureRecognizers] and
+/// [VerticalDragGestureRecognizer] are added to a team with a captain, the
+/// captain allows [AndroidView] to know when any gestures in the team has been
+/// recognized (or all other arena members are out), once the captain wins the
+/// gesture is forwarded to the Android view.
```
[the remark this answers is not in the document]
Comment (amirh) [reply packages/flutter/lib/src/gestures/team.dart]: done

**Comment by fkorotkov** [comment]
@Hixie seems Anka just tries to schedule the VM for 2 hours and then returned Error status. Unfortunately there were no useful logs but I just added some additional logs when it will happen again. Sorry about the inconvenience.

**Pull request merged**